scripts/convert_csv_to_json.py

```diff
@@ -7,40 +7,44 @@
 
 def generate_json_files():
 
-    source_directory = './source-data/gb'
-    output_directory = './data/gb'
+    source_root = './source-data'
+    output_root = './data'
 
-    for source_file_name in os.listdir(source_directory):
+    for source_directory, dirs, files in os.walk(source_root):
 
-        source_file_location = os.path.join(source_directory, source_file_name)
+        source_directory_name = source_directory.split(os.path.sep)[-1]
 
-        if os.path.isfile(source_file_location):
+        for source_file_name in os.listdir(source_directory):
 
-            source_file = open(source_file_location, 'r')
+            source_file_location = os.path.join(source_directory, source_file_name)
 
-            output_file_name = source_file_name.replace('.csv', '.json')
-            output_file_location = os.path.join(output_directory, output_file_name)
+            if os.path.isfile(source_file_location):
 
-            json_data = []
+                source_file = open(source_file_location, 'r')
 
-            with open(source_file_location, newline='') as csv_file:
-                for row in csv.DictReader(csv_file):
-                    # Append the data item as a json key-value if not an empty string
-                    json_data.append({k: v for k, v in row.items() if v})
-            try:
-                with open(output_file_location, 'w', encoding='utf8') as f:
-                    json.dump(json_data, f, indent=3, ensure_ascii=False)
-            except FileNotFoundError:
-                print(
-                    f"Output could not be written to {output_file_name}\nDoes the output folder {output_directory} exist?"
-                )
-                return
+                output_file_name = source_file_name.replace('.csv', '.json')
+                output_file_location = os.path.join(output_root, source_directory_name, output_file_name)
```

Comment:
Given that we know whether a directory will contain folders or files, using `listdir` rather than `walk` will stop the "directory is not a regular file" message being printed. Something like:

```python
for source_directory in os.listdir(source_root):
        
        source_directory_name = os.path.join(source_root, source_directory)

        for source_file_name in os.listdir(source_directory_name):

            source_file_location = os.path.join(source_directory_name, source_file_name)

            if os.path.isfile(source_file_location):

                source_file = open(source_file_location, 'r')

                output_file_name = source_file_name.replace('.csv', '.json')
                output_file_location = os.path.join(output_root, source_directory, output_file_name)
```